OneSignalSDK/unittest/src/test/java/com/test/onesignal/InAppMessagingUnitTests.java

```diff
@@ -616,4 +616,19 @@ public void testOnMessageWasShown() throws Exception {
         assertEquals(1, iamImpressionRequest.payload.get("device_type"));
         assertEquals(true, iamImpressionRequest.payload.get("first_impression"));
     }
+
+    @Test
+    public void testOnPageChanged() throws Exception {
+        threadAndTaskWait();
+
+        OneSignalPackagePrivateHelper.onPageChanged(message, InAppMessagingHelpers.buildTestPageJson());
```

Comment:
is ok to test without mocking the IAM display?

Reply:
Yes we test without mocking the iam display for clicks as well and they should be entirely independent